app/src/features/reports/CompareMapNavReport.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from "react";
+import { Grid } from "@material-ui/core";
+import UsaChloroplethMap from "../charts/UsaChloroplethMap";
+import TableChart from "../charts/TableChart";
+import WithDatasets from "../../utils/WithDatasets";
+import useDatasetStore from "../../utils/useDatasetStore";
+import variableProviders from "../../utils/variableProviders";
+import { Breakdowns } from "../../utils/Breakdowns";
+import VariableProvider from "../../utils/variables/VariableProvider";
+import { STATE_FIPS_MAP } from "../../utils/Fips";
+import Alert from "@material-ui/lab/Alert";
+import Breadcrumbs from "@material-ui/core/Breadcrumbs";
+import Link from "@material-ui/core/Link";
+import Typography from "@material-ui/core/Typography";
+
+interface Geo {
+  fips: number;
+  name: string;
+}
+
+function Map(props: {
+  fipsGeo: number;
+  data: Record<string, any>[];
+  updateGeoCallback: Function;
+}) {
+  function getStateGeoFromPropFips() {
+    return props.fipsGeo === 0
+      ? undefined
+      : { fips: props.fipsGeo, name: STATE_FIPS_MAP[props.fipsGeo] };
+  }
+
+  const [state, setState] = useState<Geo | undefined>(
+    getStateGeoFromPropFips()
+  );
+  const [county, setCounty] = useState<Geo | undefined>();
+
+  useEffect(() => {
+    props.updateGeoCallback(state ? state.fips : 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state]);
+
+  useEffect(() => {
+    setState(getStateGeoFromPropFips());
+    setCounty(undefined);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.fipsGeo]);
+
+  const signalListeners: any = {
+    click: (...args: any) => {
+      // Is this a county or a state??
+      const clickedData = args[1];
+      console.log(clickedData);
+      if (clickedData.id.length === 2) {
+        setState({ fips: clickedData.id, name: clickedData.properties.name });
+      } else {
+        setCounty({ fips: clickedData.id, name: clickedData.properties.name });
+      }
+    },
+  };
+
+  let dataset =
+    state === undefined
+      ? props.data
+      : props.data.filter((r) => r.state_fips_code === state.fips.toString());
+
+  return (
+    <div>
+      <Breadcrumbs aria-label="breadcrumb">
+        {state && (
+          <Link
+            color="inherit"
+            onClick={() => {
+              setState(undefined);
+              setCounty(undefined);
+            }}
+          >
+            United States
+          </Link>
+        )}
+        {!state && <Typography color="textPrimary">United States</Typography>}
```

Comment:
Maybe pull this out into a `GeographyBreadcrumb` component with `text`, `onClick`, and `isClickable` props and you can use it here and below?

Reply:
oh good idea